### PR TITLE
feat: add support for GPU VHD

### PR DESCRIPTION
**Pull request by bowang-666**
Add image config for GPU VHD, also make NvidiaEnabledSKUs accessible publicly so RP can use it to generate error message.

### CODE REVIEW

File: pkg/api/azenvtypes.go [hunk 1]
```diff
@@ -173,6 +173,15 @@ var (
 		ImageVersion:   "2020.05.13",
 	}
 
+	// AKSGPU1804OSImageConfig is the AKS image with GPU drivers and device plugin based on Ubuntu 18.04-LTS.
+	// TODO: Update sku and version after GPU image is actually published
+	AKSGPU1804OSImageConfig = AzureOSImageConfig{
+		ImageOffer:     "aks",
+		ImageSku:       "aks-gpu-1804-202005",
```
Comment (xuto2) [inline pkg/api/azenvtypes.go:180]: can we follow the pattern aks-ubuntu-1804-202005, maybe call it aks-ubuntu-1804-gpu-202005?

File: pkg/api/azenvtypes.go [hunk 2]
```diff
@@ -221,6 +230,7 @@ var (
 			AKSUbuntu1804:     AKSUbuntu1804OSImageConfig,
 			AKS1804Deprecated: AKSUbuntu1804OSImageConfig, // for back-compat
 			ACC1604:           ACC1604OSImageConfig,
+			AKSGPU1804:        AKSGPU1804OSImageConfig,
```
Comment (xuto2) [inline pkg/api/azenvtypes.go:233]: We should also update
AzureUSGovernmentCloudSpec 
AzureChinaCloudSpec

File: pkg/api/const.go [hunk 3]
```diff
@@ -36,6 +36,7 @@ const (
 	AKSUbuntu1604     Distro = "aks-ubuntu-16.04"
 	AKSUbuntu1804     Distro = "aks-ubuntu-18.04"
 	ACC1604           Distro = "acc-16.04"
+	AKSGPU1804        Distro = "aks-gpu-18.04"
```
Comment (xuto2) [inline pkg/api/const.go:39]: nit: AKSUbuntu1804GPU

File: pkg/api/azenvtypes.go (hunk 1 above)
Comment (bowang-666) [reply pkg/api/azenvtypes.go]: made it aks-ubuntu-gpu-1804 so numbers go in the end

File: pkg/api/azenvtypes.go (hunk 2 above)
Comment (bowang-666) [reply pkg/api/azenvtypes.go]: added. btw I don't see ACC1604 in USGov and China Cloud, are they not enabled?

File: pkg/api/const.go (hunk 3 above)
Comment (bowang-666) [reply pkg/api/const.go]: made them AKSUbuntuGPU1804

File: pkg/api/azenvtypes.go (hunk 2 above)
Comment (xuto2) [reply pkg/api/azenvtypes.go]: we don't plan to support 1604 for gen2, since we want to move to 1804

File: pkg/api/common/helper.go [hunk 4]
```diff
@@ -14,6 +14,44 @@ import (
 	validator "gopkg.in/go-playground/validator.v9"
 )
 
+var (
+	/* If a new GPU sku becomes available, add a key to this map, but only if you have a confirmation
+	   that we have an agreement with NVIDIA for this specific gpu.
+	*/
+	NvidiaEnabledSKUs = map[string]bool{
```
Comment (xuto2) [inline pkg/api/common/helper.go:21]: For future reference - we don't want to diverge from aks-e code when absolutely necessary.

**Pull request merged**